src/Containers.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import Listing from '../lib/cockpit-components-listing.jsx';
+import cockpit from 'cockpit';
+
+import ContainerDetails from './ContainerDetails.js';
+const _ = cockpit.gettext;
+
+class Containers extends React.Component {
+    constructor(props) {
+	super(props);
+	console.log(props.containers);
+	this.renderRow = this.renderRow.bind(this);
+	this.getContainerState = this.getContainerState.bind(this);
+    }
+
+    navigateToContainer (container) {
+	cockpit.location.go([ container.ID ]);
+    }
+    //TODO
+    deleteContainer(event, container) {
+	return undefined;
+    }
+    //TODO
+    stopContainer(container) {
+	return undefined;
+    }
+    //TODO
+    startContainer(container) {
+	return undefined;
+    }
+
+    //TODO
+    restartContainer(container) {
+	return undefined;
+    }
+
+    getContainerState(state) {
+	if (state.Status)
+	    return state.Status;
+	if (state.Running)
+	    return "running";
+	if (state.Paused)
+	    return "paused";
+	if (state.Restarting)
+	    return "restarting";
+	if (state.FinishedAt && state.FinishedAt.indexOf("0001") === 0)
+	    return "created";
+	return "exited";
+    }
+    renderRow(container) {
+	let isRunning = !!container.State.Running;
+	//TODO: check container.Image == container.ImageID
+	let image = container.ImageName;
+	let state = this.getContainerState(container.State);
+
+	let columns = [
+	    { name: container.Name.replace(/^\//, ''), header: true },
+	    image,
+	    // util.render_container_cmdline(container),
+	    container.Config.Entrypoint === "" ? container.Config.Cmd.join(" "):container.Config.Cmd.join(""),
+	    // TODO: CpuUsage
+	    "",
+	    //
+	    // TODO: MemoryUsage, MemoryLimit
+	    "",
+	    state,
+
+	];
+	let tabs = [{
+	    name: _("Details"),
+	    renderer: ContainerDetails,
+	    data: { container: container }
+	}];
+
+	let startStopActions = [];
+	if (isRunning)
+	    startStopActions.push({ label: _("Stop"), onActivate: this.stopContainer.bind(this, container) });
```

Comment:
Don't bind methods in a `render` function. It's a more common (and efficient?) style to write something like

    onActivate: () => this.stopContainer(container)